webapp/src/auth.ts

```diff
@@ -102,15 +103,27 @@ export async function loginAsync(idp: pxt.IdentityProviderId, persistent: boolea
  * Sign out the user and clear the auth token cookie.
  */
 export async function logout() {
-    if (!hasIdentity()) return;
+    if (!hasIdentity()) { return; }
 
     pxt.tickEvent('auth.logout');
 
-    const wasLoggedIn = loggedIn();
-    await apiAsync('/api/auth/logout');
+    // backend will clear the cookie token and pass back the provider logout endpoint.
+    const result = await apiAsync(core.stringifyQueryString(
+        '/api/auth/logout', {
+        // Where to end up after logging out from provider.
+        redirect_uri: window.location.origin
+    }));
+
+    // Clear header token so we can no longer make authenticated requests.
+    pxt.storage.removeLocal(AUTH_TOKEN);
+
+    // Update state and UI to reflect logged out state.
     clearState();
-    if (wasLoggedIn) {
-        core.infoNotification(lf("Signed out"));
+
+    // Redirect to provider's logout endpoint. If all goes well there, we'll
+    // be redirected back here.
+    if (result.resp?.logout_uri) {
+        window.location.href = result.resp.logout_uri;
```

Comment:
I know other sites like Tynker do this, but it still seems wrong to me to log out of the provider when the user just said "logout" on our site. E.g. logout from MakeCode shouldn't mean logout from MSFT and GitHub too.
But I'm not going to die on that hill

Reply:
I think you're right that we shouldn't sign out of the upstream provider. This will sign the user out from our app on all devices. If we want to support that, it should be a choice: "Sign out from: [Just here] [Everywhere]"